CenterForOpenScience/osf.io
# [ENG-1366] Add cookie setting to track logged out users

## Purpose

Make sure user information is recorded for preprints as part of our Sloan study.

## Changes

- adds a cookie to identify logged out users
- adds a cookie value to preprints metrics info

## QA Notes

Dev test only

The best way to test this is to create a preprint and then view/download it* and then check in the shell that the metrics were recorded correctly. https://elasticsearch-py.readthedocs.io/en/master/
`PreprintDownload.search().filter('exists', field='cookie').execute()` will return all downloads with a cookie. 

* with a user who isn't a contributor!

* We should verify that there is existing Impact data on staging before rolling this out, to verify that it transitions properly to having the new, additional data when this is added.

## Documentation

Not user facing, no new docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1366
